ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/transaction/arjunacore/jca/TransactionImporterImple.java

```diff
@@ -167,11 +181,19 @@ public SubordinateTransaction getImportedTransaction(Xid xid)
 		if (xid == null)
 			throw new IllegalArgumentException();
 
-		SubordinateTransaction tx = _transactions.get(new SubordinateXidImple(xid));
+		AtomicReference<TransactionImple> holder = _transactions.get(new SubordinateXidImple(xid));
 
-		if (tx == null)
+		if (holder == null)
 			return null;
-		
+
+		SubordinateTransaction tx = holder.get();
+
+		if (tx == null) {
+			// another thread must be in the process of importing this transaction
+			// TODO the solution does not provide a mechanism to wait for the other thread to update the holder
```

Comment:
This solution does not provide a mechanism to wait for the other thread to update the holder